File: Sorting_Algorithm/Bubble_sort.java
package Sorting_Algorithm;
import java.util.Arrays;
public class Bubble_sort {
  public static void main(String[]args){
    int[] arr={9,8,7,6,5,4,3,2,1};
    for(int i=0;i<arr.length;i++) {
			for(int j=0;j<arr.length-i-1;j++) {
				if(arr[j]>arr[j+1]) {
					int temp=arr[j];
					arr[j]=arr[j+1];
					arr[j+1]=temp;
				}
			}
		}
    System.out.println(Arrays.toString(arr));
  }
}

File: Sorting_Algorithm/C++/bubble-sort.cpp
#include <iostream>
using namespace std;

void bubbleSort(int *, int);

int main()
{
    int n; // size
    cin >> n;
    int arr[n];

    for (int i = 0; i < n; i++) // input elements
        cin >> arr[i];

    bubbleSort(arr, n);

    for (auto i : arr) // output
        cout << i << " ";

    cout << endl;

    return 0;
}

void bubbleSort(int *arr, int n)
{
    for (int pass = 1; pass <= n - 1; pass++) // running passes
    {
        bool flag = false; // no swapping happened

        for (int i = 0; i <= n - pass - 1; i++) // traversing
        {
            if (arr[i] > arr[i + 1]) // for ascending order sorting
            {
                flag = true; // swapping happened

                // swap
                int temp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = temp;
            }
        }

        if (!flag) // if (false == flag) i.e. if there is no swapping
            break; // no need for any more passes
    }
}